### PR TITLE
Ensure that Numpy is installed before it is referenced.

### PR DESCRIPTION
**Pull request by paul-romlow**
When installing to an environment that does not have Numpy installed, the build fails. This waits to reference Numpy until after the setup_requires section has been resolved.

### CODE REVIEW

**Comment by nikolasibalic** [comment]
HI @paul-romlow 

Thank you for contribution. I have just one question trying to understand use-case. Usually the dependencies that are needed are installed via `requirements.txt`.
This is done automatically when you do pip install. How did you do installation to run into this problem? Maybe you had some reason why you couldn't run pip install ?

**Comment by paul-romlow** [comment]
We use Pants for a build system which creates a virtual environment first and then it builds the source (Linux, so no binary available) before installing all of the other requirements.
https://www.pantsbuild.org/

**Comment by paul-romlow** [comment]
If this PR is not acceptable to you, would it be possible for you to publish a wheel (bdist_wheel) for amd64 Linux as is done with Windows and MacOS? I'd be happy to help with getting that set up, please let me know.

Thanks!

File: setup.py [hunk 1]
```diff
@@ -50,6 +58,7 @@
         "Topic :: Scientific/Engineering :: Physics",
         "Development Status :: 5 - Production/Stable",
     ],
+    setup_requires=["numpy>=1.16.0"],
```
Comment (nikolasibalic) [inline setup.py:61]: Can we change this to package named "oldest-supported-numpy" ?
That one will automatically pick oldest supported Numpy for corresponding Python version.
See [here](https://github.com/nikolasibalic/ARC-Alkali-Rydberg-Calculator/issues/100#issuecomment-1036258981) for the reasons.
`setup_requires=["oldest-supported-numpy"],`

**Review by nikolasibalic — CHANGES_REQUESTED** [review]

**Comment by nikolasibalic** [comment]
@paul-romlow Please check if the above change is fine with you. If so, commit update, and I will accept this PR and release a new version.

**Pull request merged**